lib/Demangling/Remangler.cpp

```diff
@@ -1954,6 +1957,47 @@ void Remangler::mangleReabstractionThunkHelperWithSelf(Node *node) {
   Buffer << "Ty";
 }
 
+// SWIFT_ENABLE_TENSORFLOW
+void Remangler::mangleAutoDiffParameterIndices(Node *node) {
+  Buffer << 'p';
+  for (unsigned i = 0, n = node->getNumChildren(); i != n; ++i) {
+    auto child = node->getChild(i);
+    Buffer << child->getIndex();
+    if (i != n - 1)
+      Buffer << '_';
+  }
+}
+
+void Remangler::mangleAutoDiffResultIndex(Node *node) {
+  Buffer << 'r' << node->getIndex();
+}
+
+void
+Remangler::mangleAutoDiffAssociatedFunctionHelper(Node *node, StringRef op) {
+  assert(node->getNumChildren() == 3);
+  mangleChildNode(node, 0); // original function
+  Buffer << op;
+  mangleChildNode(node, 1); // wrt parameter indices
+  mangleChildNode(node, 2); // result index
```

Comment:
Add a TODO for mangling generic signature?

Reply:
Done in https://github.com/apple/swift/pull/26624/commits/e8a3e55f190cad3d0344f164ef5e440816015173.